File: app/Quotes/bs/src/widgets/wrule.ml
(*  Copyright 30-Jul-2018 ºDeme
    GNU General Public License - V3 <http://www.gnu.org/licenses/>
*)

open Ui

type t

let mk width color title =
  let w = string_of_int width in
  q "table" [Klass "main"; Style {j|color:$color;|j}][
    q "tr" [][
      q "td" [Style {j|width:$(w)px;|j}][q "hr"[][]];
      q "td" [Style "width:5px;white-space: nowrap;"; Html title][];
      q "td" [Style {j|width:$w;|j}][q "hr"[][]]]]

let mk_big = mk 50 "#101010"

let mk_small = mk 20 "#808080"

File: app/MultiMarket/js/src/wgs/Wrule.js
// Copyright 11-Sept-2018 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

import Ui from "../dmjs/Ui.js";

const $ = e => Ui.$(e);

/** Wrule widget. */
export default class Wrule {

  static mk (width, color, title) {
    return $("table").klass("main").style(`color:${color}`)
      .add($("tr")
        .add($("td").style(`width:${String(width)}px;`).add($("hr")))
        .add($("td").style("width:5px;white-space: nowrap;").html(title))
        .add($("td").add($("hr"))))
    ;
  }

  static mkBig (title) {
    return Wrule.mk(50, "#101010", title);
  }

  static mkSmall (title) {
    return Wrule.mk(20, "#808080", title);
  }

}
